compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt

```diff
@@ -565,3 +564,28 @@ private fun Binding.key(module: ModuleDescriptor): String {
 
   return fqName + allArguments
 }
+
+private class MergedClassKey(
```

Comment:
`data class` doesn't suffice for your needs?

Reply:
@vRallev I think you missed this comment

Reply:
Ah, sorry. Notice that I ignore `clazz` in the methods and use its FqName instead. That wouldn't work with a `data class`.